modules/repofiles/upload.go

```diff
@@ -55,6 +55,23 @@ func UploadRepoFiles(repo *models.Repository, doer *models.User, opts *UploadRep
 		return fmt.Errorf("GetUploadsByUUIDs [uuids: %v]: %v", opts.Files, err)
 	}
 
+	names := make([]string, len(uploads))
+	infos := make([]uploadInfo, len(uploads))
+	for i, upload := range uploads {
+		// Check file is not lfs locked
+		filepath := path.Join(opts.TreePath, upload.Name)
+		lfsLock, err := repo.GetTreePathLock(filepath)
```

Comment:
Sorry I missed this earlier - you should only check LFS Locks if `setting.LFS.Server`. You need to wrap this code in a check.

Reply:
It's actually in GetTreePathLock, but I am not sure it is a good design.

Reply:
Ah ok just add a comment above this line to say that if LFS is off then this will return nil.

Reply:
There should probably be a code refactoring regarding GetTreePathLock, but I went for the easy way now and only added a code comment.